src/ParametricCliqStateMachine.jl

```diff
@@ -0,0 +1,337 @@
+  Z::Vector{Float64}
+  inferdim::Float64
+end
+
+#TODO pack and unpack
+getSample(s::ParametricMsgPrior, N::Int=1) = (s.Z, )
+
+struct PackedParametricMsgPrior <: PackedInferenceType where T
+  Z::String
+  inferdim::Float64
+end
+
+function convert(::Type{PackedParametricMsgPrior}, d::ParametricMsgPrior)
+  PackedParametricMsgPrior(string(d.Z), d.inferdim)
+end
+function convert(::Type{ParametricMsgPrior}, d::PackedParametricMsgPrior)
+  ParametricMsgPrior([0.0], d.inferdim)
+end
+
+#TODO move to TreeBasedInitialization.jl
+function addMsgFactors!(subfg::G,
+                        msgs::ParametricBeliefMessage)::Vector{DFGFactor} where G <: AbstractDFG
+  # add messages as priors to this sub factor graph
+  msgfcts = DFGFactor[]
+  svars = DFG.getVariableIds(subfg)
+  for (msym, belief) = (msgs.belief)
+    if msym in svars
+      # TODO prior missing manifold information? is it not available in variable?
+      # TODO add MsgPrior parametric type
+      @warn "TODO I just packed in MvNormal and ignored cov"
+      msgPrior =  MsgPrior(MvNormal(belief.val[:,1], 1), belief.inferdim)
+      fc = addFactor!(subfg, [msym], msgPrior, autoinit=false)
+      push!(msgfcts, fc)
+    end
+  end
+  return msgfcts
+end
+
+"""
+    $SIGNATURES
+
+Notes
+- Parametric state machine function nr. 2
+"""
+function waitForUp_ParametricStateMachine(csmc::CliqStateMachineContainer)
+
+  infocsm(csmc, "Par-2, wait for up messages of needed")
+
+  setCliqDrawColor(csmc.cliq, "purple") #TODO don't know if this is correct color
+  csmc.drawtree ? drawTree(csmc.tree, show=false, filepath=joinpath(getSolverParams(csmc.dfg).logpath,"bt.pdf")) : nothing
+
+  for e in Graphs.out_edges(csmc.cliq, csmc.tree.bt)
+    @info "$(csmc.cliq.index): take! on edge $(e.index)"
+    # Blocks until data is available.
+    beliefMsg = take!(csmc.tree.messages[e.index].upMsg)
+    @info "$(csmc.cliq.index): Belief message recieved with status $(beliefMsg.status)"
+    #TODO save up message (and add priors to cliqSubFg) gaan eers in volgende stap kyk hoe dit werk
+    #kies csmc vir boodskappe vir debugging, dis 'n vector een per kind knoop
+    push!(csmc.parametricMsgsUp, beliefMsg)
+  end
+
+  return solveUp_ParametricStateMachine
+end
+
+function Graphs.in_edges(vert::V, gr::GenericIncidenceList{V, Edge{V}, Vector{V}}) where {V}
+  inclist = gr.inclist
+  targid = vert.index
+  inlist = Edge{V}[]
+  for edgelist in inclist
+    for ed in edgelist
+      if ed.target.index == targid
+        push!(inlist, ed)
+      end
+    end
+  end
+  return inlist
+end
+
+"""
+    $SIGNATURES
+
+Notes
+- Parametric state machine function nr. 3
+"""
+function solveUp_ParametricStateMachine(csmc::CliqStateMachineContainer)
+
+  infocsm(csmc, "Par-3, Solving Up")
+
+  setCliqDrawColor(csmc.cliq, "red")
+  csmc.drawtree ? drawTree(csmc.tree, show=false, filepath=joinpath(getSolverParams(csmc.dfg).logpath,"bt.pdf")) : nothing
+
+  # is dit nodig om vir parametric die factors by te sit?
```

Comment:
in the current mmisam solution the factors symbols are recorded in the BayesTreeNodeData for each clique, but actual factors are only fetched from the distributed factor graph object when the cliqSubFg is created.  Perhaps okay to follow that pattern here too for the time being...

Reply:
Good to know, I think that is why I couldn't remove the duplicate to test if it is a bug.